### PR TITLE
feat: allow relative asset paths & example

### LINKED ISSUE #128: Relative asset paths
### Discussed in https://github.com/digita-ai/semcom/discussions/78
<div type='discussions-op-text'>

<sup>Originally posted by **woutermont** May 11, 2021</sup>
Do we have any idea how semcom will/should behave when the client requests a component whose HTML contains a relative path to an asset (e.g. images)? Do we require all assets (even large ones) to be bundled in a single JS file?

</div>

┆Issue is synchronized with this [Wrike Item](https://www.wrike.com/open.htm?id=728634962) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by jonasbovyn**
fixes #128

### CODE REVIEW

File: packages/semcom-components/lib/components/gender.component.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  static get styles() {
+
+    return [
+      css`
+        
+      `,
+    ];
+
+  }
```
Comment (woutermont) [inline packages/semcom-components/lib/components/gender.component.ts:53]:
```suggestion
  static get styles() { return []; }
```

File: packages/semcom-components/lib/components/gender.component.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  render() {
+
+    const imgUrl = new URL(`../assets/gender/${this.gender}.png`, import.meta.url);
```
Comment (woutermont) [inline packages/semcom-components/lib/components/gender.component.ts:57]: So, what made this work in the end?

File: packages/semcom-components/lib/components/gender.component.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    }
+
+    const genderPrefix = 'http://digita.ai/voc/gender#';
+
+    const store = new Store(event.detail.data);
+
+    this.gender = store.getQuads(null,  new NamedNode(`${genderPrefix}type`), null, null)[0]?.object.value.split('#')[1];
+
+  }
```
Comment (woutermont) [inline packages/semcom-components/lib/components/gender.component.ts:30]:
```suggestion
  @property() gender = 'Unknown';
  
  private genderPredicate = new NamedNode('http://digita.ai/voc/gender#type');

  /**
   * Handles a response event. Can be used to update the component's properties based on the data in the response.
   *
   * @param event The response event to handle.
   */
  handleResponse(event: ComponentResponseEvent): void {

    if (!event || !event.detail || !event.detail.data) {

      throw new Error('Argument event || !event.detail || !event.detail.quads should be set.');

    }

    const store = new Store(event.detail.data);

    this.gender = store.getQuads(null, genderPredicate, null, null)[0]?.object.value.split('#')[1];

  }
```

**Review by woutermont — CHANGES_REQUESTED** [review]
Lower `target` ECMAScript versions in Vite and TS configs do serve a purpose (browser compatibility), so we should try to keep as much of them not at the alter version. (For `module` settings, this doesn't matter that much, normally).

File: packages/semcom-components/lib/components/gender.component.ts (hunk 2 above)
Comment (jonasbovyn) [reply packages/semcom-components/lib/components/gender.component.ts:57]: I think a bug in "esnext", I changed es2017 to esnext, and inspected the js files of the components, and saw that import.meta was being replaced by a variable "e", which had been assigned the value {} to it

It was fixed when I changed it to es2020. However, according to what I could find, esnext should've worked, but it didn't

**Pull request merged**